### PR TITLE
UML-1518 - PHP Linting with PHP Code Sniffer

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
# Purpose

Give developers feedback on php linting issues

NOTE: This is using a fork of the action-php-codesniffer Github Action to solve a problem of standards files being ignored

Fixes UML-1518

## Approach

- add a github action to run PHP CodeSniffer
- Use a matrix to run linting for each app path

## Learning

~https://github.com/marketplace/actions/action-php-codesniffer~
https://github.com/tinovyatkin/action-php-codesniffer
https://github.com/thenabeel/action-phpcs

## Checklist

* [ ] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] The product team have tested these changes

### CODE REVIEW

File: .github/workflows/php-linting.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - ".github/workflows/php-linting.yml"
+
+permissions:
+  actions: read
+  checks: read
+  contents: none
+  deployments: none
+  issues: none
+  packages: none
+  pull-requests: read
```
Comment (andrewpearce-digital) [inline .github/workflows/php-linting.yml:21]:
```suggestion
  pull-requests: write
```

File: service-front/app/public/index.php [hunk 2]
```diff
@@ -3,12 +3,11 @@
...
 
 // Delegate static file requests back to the PHP built-in webserver
-if (PHP_SAPI === 'cli-server' && $_SERVER['SCRIPT_FILENAME'] !== __FILE__) {
+if (PHP_SAPI==='cli-server' && $_SERVER['SCRIPT_FILENAME'] !== __FILE__) {
     return false;
 }
 
-chdir(dirname(__DIR__));
-require 'vendor/autoload.php';
+chdir(dirname(__DIR__));require 'vendor/autoload.php';
```
Comment (andrewpearce-digital) [inline service-front/app/public/index.php:10]:
```suggestion
chdir(dirname(__DIR__));require 'vendor/autoload.php'; 
```

File: .github/workflows/php-linting.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      matrix:
+        service_path: [ 'service-front', 'service-api' ]
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0 # important!
```
Comment (andrewpearce-digital) [inline .github/workflows/php-linting.yml:41]:
```suggestion
        fetch-depth: 2 # important!
```

File: .github/workflows/php-linting.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      matrix:
+        service_path: [ 'service-front', 'service-api' ]
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 2 # important!
```
Comment (andrewpearce-digital) [inline .github/workflows/php-linting.yml:41]:
```suggestion
```

File: .github/workflows/php-linting.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    strategy:
+      matrix:
+        service_path: [ 'service-front', 'service-api' ]
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
```
Comment (andrewpearce-digital) [inline .github/workflows/php-linting.yml:40]:
```suggestion
```

**Comment by cooperaj** [comment]
Not sure it's picking up the config file correctly as it's doing linting operations from standards we don't use.

**Comment by andrewpearce-digital** [comment]
> Not sure it's picking up the config file correctly as it's doing linting operations from standards we don't use.

You're right, there's an issue on the action showing some incompatibility with blame and standards options.

I've replaced the phpcs action and it's now respecting the standards

**Comment by andrewpearce-digital** [comment]
Action now provides inline comments
image

**Review by johnnolan — APPROVED** [review]

**Review by cooperaj — APPROVED** [review]

**Pull request merged**